### ISSUE TITLE
Add more threading supoprt

### ISSUE THREAD

**Issue by bkamins** [JuliaData/DataFrames.jl#2626]
After https://github.com/JuliaData/DataFrames.jl/pull/2622 is merged add threading support to:
* `getindex`
* simple cases of `select`
* joins

In general the idea is that we can copy/create different columns of a data frame in parallel if we do not do any operations on them.

**Comment by bkamins** [comment]
I am closing this in favor of specific issues/PRs:
https://github.com/JuliaData/DataFrames.jl/issues/2660
https://github.com/JuliaData/DataFrames.jl/issues/2659
https://github.com/JuliaData/DataFrames.jl/issues/2658
https://github.com/JuliaData/DataFrames.jl/pull/2647
